hack/update-vendor.sh

```diff
@@ -233,7 +233,9 @@ for repo in $(kube::util::list_staging_repos); do
   (
     echo "=== tidying k8s.io/${repo}"
     cd "staging/src/k8s.io/${repo}"
+    sed -E -i 's/^go 1\.(.*)$/go 1.21.0/' go.mod
     go mod tidy -v
+    sed -i '/^toolchain /d' go.mod
```

Comment:
needing to do fixup to remove things `go mod tidy` added is not great ... running `go mod tidy` again should be a no-op when we're done with `hack/update-vendor.sh`

Reply:
@liggitt i agree, not sure how to control `go mod tidy`'s behavior here

Reply:
https://go-review.git.corp.google.com/c/go/+/657178

Reply:
https://go.dev/cl/657178

Reply:
https://github.com/golang/go/commit/a68bf75d3402412a1946fe1df67f57ca923f1507